src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -329,8 +339,7 @@ public virtual Task InvokeAsync(Func<Task> workItem)
             // This is for example when we run on a system with a single thread, like WebAssembly.
             if (_dispatcher == null)
             {
-                workItem();
-                return Task.CompletedTask;
+                return workItem();
```

Comment:
This was a bug that broke an E2E test. I'm not sure how we didn't detect this somehow.